### PR TITLE
Add optional sys_info field to events

### LINKED ISSUE #152: Add ability to log CPU/threading info
In msquic, events all contain the current CPU ID, Process ID and Thread ID (https://github.com/microsoft/msquic/blob/main/src/plugins/wpa/DataModel/QuicEvent.cs), which allows them to do advanced debugging of threading and CPU allocation issues (https://github.com/microsoft/msquic/blob/main/docs/TSG.md#diagnosing-rss-issues).

It would potentially be interesting to have this type of metadata well-defined in qlog as well.

CC @nibanks

### PR DESCRIPTION
**Pull request by LPardue**
Fixes #152 by adding a field to capture information related to CPUs or
threads.

### CODE REVIEW

**Comment by LPardue** [comment]
cc @nibanks

File: draft-ietf-quic-qlog-main-schema.md [hunk 1]
```diff
@@ -891,6 +892,24 @@ instead of logging the "group_id" field with an identical value for each event
...
+different combinations thereof. Each field is optional to support deployment
+diversity.
+
+Definition:
+
+~~~ cddl
+SystemInformation = {
+  ? processor_id: uint32
+  ? process_id: uint32
+  ? thread_id: uint32
```
Comment (marten-seemann) [inline draft-ietf-quic-qlog-main-schema.md:909]: This only works if the QUIC stack is single-threaded. Is that a reasonable assumption?

**Review by marten-seemann — APPROVED** [review]

File: draft-ietf-quic-qlog-main-schema.md (hunk 1 above)
Comment (LPardue) [reply draft-ietf-quic-qlog-main-schema.md:909]: How so? From the discussion on #152, it seemed fine to have events in the same logical trace that had different thread IDs?

If it seems I'm preventing that in this propsal, I'd like to fix it up so all suggestions welcome.

File: draft-ietf-quic-qlog-main-schema.md [hunk 2]
```diff
@@ -891,6 +892,24 @@ instead of logging the "group_id" field with an identical value for each event
 instance, this field is typically logged once in "common_fields", see
 {{common-fields}}.
 
+### system_info
+
+The "sys_info" field can be used to record system-specific details related to an
```
Comment (rmarx) [inline draft-ietf-quic-qlog-main-schema.md:897]:
```suggestion
The "system_info" field can be used to record system-specific details related to an
```

consistency with line 571 above

**Review by rmarx — APPROVED** [review]
Minus the small nit.

File: draft-ietf-quic-qlog-main-schema.md (hunk 1 above)
Comment (rmarx) [reply draft-ietf-quic-qlog-main-schema.md:909]: I agree with Lucas here, as also discussed on the original issue.

File: draft-ietf-quic-qlog-main-schema.md (hunk 2 above)
Comment (LPardue) [reply draft-ietf-quic-qlog-main-schema.md:897]: oops

**Pull request merged**